### DISCUSSION TITLE
Failed to upload archive and Failed to publish

### DISCUSSION THREAD

**Discussion by panwar001** [codemagic-ci-cd/codemagic-docs#2276] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

I am trying to publish build ipa to appstore. Current app version is 1.3.4+4, where as codemagic compiler is giving invalid version error 1.3.3 . Not getting what i am doing wrong, I have tried multiple app version in pubspec.yaml, but every time same error.

Codemagic.yaml Snapshot- 
<img width="1168" alt="Screenshot 2023-12-18 at 3 24 19 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/8311777/ff0b820c-848d-4ed9-a2a8-e6d52b70b66d">

Codemagic Logs Snapshot - 
<img width="1097" alt="291229405-a172fe92-3a61-422a-8788-72af159d02d4" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/8311777/230d7e11-dbbb-4e8d-9ddd-7b66f6adf12c">

Flutter Pubspec.yaml snapshot
<img width="894" alt="Screenshot 2023-12-18 at 3 32 23 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/8311777/204dcb4f-301b-44aa-a74a-ba3bbab51878">

Any help would be appreciated! 

### Expected results

To publish build ipa to appstore

### Actual results

Error - 
[{"message":"Asset validation failed","userInfo":{"NSUnderlyingError":"Error Domain=IrisAPI Code=-19241 \"Asset validation failed\" UserInfo={status=409, detail=Invalid Version. The build with the version “1.3.3” can’t be imported because a later version has been closed for new build submissions. Choose a different version number., id=61886e80-c792-46da-b573-93b2f7d7ce86, code=STATE_ERROR.VALIDATION_ERROR.90478, title=Asset validation failed, NSLocalizedFailureReason=Invalid Version. The build with the version “1.3.3” can’t be imported because a later version has been closed for new build submissions. Choose a different version number., NSLocalizedDescription=Asset validation failed}","NSLocalizedDescription":"Asset validation failed","iris-code":"STATE_ERROR.VALIDATION_ERROR.90478","NSLocalizedFailureReason":"Invalid Version. The build with the version “1.3.3” can’t be imported because a later version has been closed for new build submissions. Choose a different version number. (ID: 61886e80-c792-46da-b573-93b2f7d7ce86)"},"code":90478}

### Build id (optional)

657e7c0a5c4512769927b871

**Comment by dtrdic** [comment, marked as answer]
hello @panwar001, try adding `agvtool new-version -all $(($BUILD_NUMBER + 1))` command which is responsible for increasing the version number in your Xcode project.

Add it before the` flutter build ipa` command

```
- name: Flutter build ipa and automatic versioning
   script: |
       agvtool new-version -all $(($BUILD_NUMBER + 1))
       flutter build ipa --release
```

Also make sure that Info.plist file has the following string along with their key values

```
<key>CFBundleShortVersionString</key>
<string>$(FLUTTER_BUILD_NAME)</string>
<key>CFBundleVersion</key>
<string>$(FLUTTER_BUILD_NUMBER)</string>
```

**Reply by panwar001** [reply]
Thanks @dtrdic !
Codemagic needs to update its documentation. This creating a lot of confusion.
